### PR TITLE
AZP/RELEASE: Build in-tree, move back to the working directory

### PR DESCRIPTION
**Pull request by Al3xR01**
## What
Switch to in-tree Deb build.

## Why ?
https://github.com/openucx/ucx/issues/8854

### CODE REVIEW

**Review by yosefe — APPROVED** [review]

File: buildlib/az-distro-release.yml [hunk 1]
```diff
@@ -67,9 +62,13 @@ jobs:
 
       - bash: |
           set -eEx
-          cd pkg-build
-          echo 10 > debian/compat   # https://www.debian.org/doc/manuals/maint-guide/dother.en.html#compat
-          sed -i 's|./contrib|../contrib|' debian/rules  # Hack
+          ./autogen.sh
+          ./contrib/configure-release --with-cuda --with-java=no
```
Comment (brminich) [inline buildlib/az-distro-release.yml:66]: why is there no need for sed hack with contrib path anymore?

Comment (yosefe) [reply buildlib/az-distro-release.yml:66]: because the build is no longer out-of-tree on ubuntu

**Review by yosefe — DISMISSED** [review]

File: buildlib/az-distro-release.yml [hunk 2]
```diff
@@ -47,14 +47,9 @@ jobs:
       - bash: |
           set -eEx
           ./autogen.sh
-          mkdir pkg-build
+          mkdir -p pkg-build
```
Comment (artemry-nv) [inline buildlib/az-distro-release.yml:50]: Seems -p is redundant here

**Review by artemry-nv — APPROVED** [review]

File: buildlib/az-distro-release.yml (hunk 2 above)
Comment (Al3xR01) [reply buildlib/az-distro-release.yml:50]: Might be superficial here, but does not harm.
I've seen at least one CI failure, where this command was executed before the parent folder was created.

**Review by yosefe — APPROVED** [review]

**Pull request merged**